Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt

```diff
@@ -0,0 +1,288 @@
+import com.example.reply.ui.utils.ReplyNavigationContentPosition
+import com.example.reply.ui.utils.ReplyNavigationType
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReplyApp(
+    windowSize: WindowSizeClass,
+    foldingDevicePosture: DevicePosture,
+    replyHomeUIState: ReplyHomeUIState,
+    closeDetailScreen: () -> Unit = {},
+    navigateToDetail: (Long) -> Unit = {}
+) {
+    /**
+     * This will help us select type of navigation and content type depending on window size and
+     * fold state of the device.
+     *
+     * In the state of folding device If it's half fold in BookPosture we want to avoid content
+     * at the crease/hinge
+     */
+    val navigationType: ReplyNavigationType
+    val contentType: ReplyContentType
+
+    when (windowSize.widthSizeClass) {
+        WindowWidthSizeClass.Compact -> {
+            navigationType = ReplyNavigationType.BOTTOM_NAVIGATION
+            contentType = ReplyContentType.LIST_ONLY
+        }
+        WindowWidthSizeClass.Medium -> {
+            navigationType = ReplyNavigationType.NAVIGATION_RAIL
+            contentType = if (foldingDevicePosture != DevicePosture.NormalPosture) {
+                ReplyContentType.LIST_AND_DETAIL
+            } else {
+                ReplyContentType.LIST_ONLY
+            }
+        }
+        WindowWidthSizeClass.Expanded -> {
+            navigationType = if (foldingDevicePosture is DevicePosture.BookPosture) {
+                ReplyNavigationType.NAVIGATION_RAIL
+            } else {
+                ReplyNavigationType.PERMANENT_NAVIGATION_DRAWER
+            }
+            contentType = ReplyContentType.LIST_AND_DETAIL
+        }
+        else -> {
+            navigationType = ReplyNavigationType.BOTTOM_NAVIGATION
+            contentType = ReplyContentType.LIST_ONLY
+        }
+    }
+
+    /**
+     * Content inside Navigation Rail/Drawer can also be positioned at top, bottom or center for
+     * ergonomics and reachability depending upon the height of the device.
+     */
+    val navigationContentPosition = when (windowSize.heightSizeClass) {
+        WindowHeightSizeClass.Compact -> {
+            ReplyNavigationContentPosition.TOP
+        }
+        WindowHeightSizeClass.Medium -> {
+            ReplyNavigationContentPosition.CENTER
+        }
+        WindowHeightSizeClass.Expanded -> {
+            ReplyNavigationContentPosition.BOTTOM
+        }
+        else -> {
+            ReplyNavigationContentPosition.TOP
+        }
+    }
+
+    ReplyNavigationWrapperUI(
+        navigationType = navigationType,
+        contentType = contentType,
+        navigationContentPosition = navigationContentPosition,
+        replyHomeUIState = replyHomeUIState,
+        closeDetailScreen = closeDetailScreen,
+        navigateToDetail = navigateToDetail
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ReplyNavigationWrapperUI(
```

Comment:
[nit] I haven't seen us use a UI suffix on composables - perhaps because all composables are UI so it doesn't need it?